pkg/services/sqlstore/migrations/ualert/channel.go

```diff
@@ -26,11 +23,56 @@ type notificationChannel struct {
 }
 
 // channelsPerOrg maps notification channels per organisation
-type channelsPerOrg map[int64]map[interface{}]*notificationChannel
+type channelsPerOrg map[int64][]*notificationChannel
 
 // channelMap maps notification channels per organisation
 type defaultChannelsPerOrg map[int64][]*notificationChannel
 
+// Union type for receiver map that contains both uid -> receiver and ID -> receiver
+type uidOrID interface{}
+
+// Start channel and alert migrations. This creates Alertmanager configs as well as migrates receivers and creates the initial root-level route.
+// Returns per org maps containing: The alertmanager config, all migrated receivers, all migrated default receivers
+func (m *migration) setupAlertmanagerConfigs() (amConfigsPerOrg, map[int64]map[uidOrID]*PostableApiReceiver, map[int64]map[string]struct{}, error) {
+	// allChannels: channelUID -> channelConfig
+	allChannelsPerOrg, defaultChannelsPerOrg, err := m.getNotificationChannelMap()
+	if err != nil {
+		return nil, nil, nil, err
+	}
+
+	amConfigPerOrg := make(amConfigsPerOrg, len(allChannelsPerOrg))
+	receiversPerOrg := make(map[int64]map[uidOrID]*PostableApiReceiver, len(allChannelsPerOrg))
+	defaultReceiversPerOrg := make(map[int64]map[string]struct{}, len(allChannelsPerOrg))
+	for orgID := range allChannelsPerOrg {
+		amConfig, ok := amConfigPerOrg[orgID]
+		if !ok {
+			amConfig = &PostableUserConfig{
+				AlertmanagerConfig: PostableApiAlertingConfig{
+					Receivers: make([]*PostableApiReceiver, 0),
+				},
+			}
+			amConfigPerOrg[orgID] = amConfig
+		}
+
+		defaultChannels := defaultChannelsPerOrg[orgID]
```

Comment:
It wasn't obvious to me, but after testing I see that it is possible for an organization to not have any default channels, should we validate this here to make it a bit more explicit? Perhaps something like:

```
		defaultChannels := []*notificationChannel{}
		defaultReceiversPerOrg[orgID] = make(map[string]struct{})
		_, ok = defaultChannelsPerOrg[orgID]
		if ok {
			// If the organization has default channels build a list of default receivers.
			defaultChannels = defaultChannelsPerOrg[orgID]

			for _, c := range defaultChannels {
				defaultReceiversPerOrg[orgID][c.Name] = struct{}{}
			}
		}
```